source/kraken/apply_disruption.cpp

```diff
@@ -401,6 +401,7 @@ struct add_impacts_visitor : public apply_impacts_visitor {
                 vj->name = mvj->get_base_vj().at(0)->name;
                 vj->headsign = mvj->get_base_vj().at(0)->headsign;
                 vj->dataset = mvj->get_base_vj().at(0)->dataset;
+                vj->dataset->vehiclejourney_list.insert(vj);
             } else {
```

Comment:
You should also do this in the "else" part, no?

And maybe also for the other functors, not only metaVJ/route, but RailSection, LineSection?

Reply:
for other functors, the `vj->dataset` is not even set... if it works for them until now, I'm not gonna change it

Reply:
But in the "else" part, dataset is provided, so adding it would probably improve erase (replace) of full trip-add (no base VJ)?
Not sure, though.

Reply:
when handling other kinds of impact, new vj's dataset is left `nullptr`, 
according to this code: https://github.com/hove-io/navitia/blob/dev/source/type/meta_vehicle_journey.cpp#L94, `erase` is not even performed...

Reply:
There was a little misunderstanding on this comment that had 2 point:
* one on the "else" part of this code, which is handled with your new commit
* the other on other functors, where erase is not performed as you mentioned
So I resolve this.